shanoir-ng-datasets/src/main/java/org/shanoir/ng/configuration/amqp/RabbitMqReceiver.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+package org.shanoir.ng.configuration.amqp;
+
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+
+import org.shanoir.ng.shared.core.model.IdName;
+import org.shanoir.ng.shared.model.Study;
+import org.shanoir.ng.shared.model.Subject;
+import org.shanoir.ng.shared.repository.StudyRepository;
+import org.shanoir.ng.shared.repository.SubjectRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author yyao
+ *
+ */
+@Configuration
+public class RabbitMqReceiver {
+	
+	private static final String STUDY_NAME_UPDATE = "study_name_update";
+	private static final String SUBJECT_NAME_UPDATE = "subject_name_update";
+	private static final Logger LOG = LoggerFactory.getLogger(RabbitMqReceiver.class);
+	
+	@Autowired StudyRepository studyRepository;
+	@Autowired SubjectRepository subjectRepository;
+	
+	private CountDownLatch latch = new CountDownLatch(1);
+	
+	public CountDownLatch getLatch() {
+		return latch;
+	}
+	
+	@RabbitListener(queues = STUDY_NAME_UPDATE)
+	public void receiveStudyNameUpdate(final String studyStr) {
+		
+		ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
I think this objectMapper may have to be obtained from Spring. Can you confirm @michaelkain  ? :)

Reply:
yes, please use autowired here